File: 80. Remove Duplicates from Sorted Array II/80. Remove Duplicates from Sorted Array II.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        n = len(nums)

        if n < 3:
            return n

        left = 2
        for right in range(2, n):
            if nums[right] != nums[left - 2]:
                nums[left] = nums[right]
                left += 1
        return left

File: 80. Remove Duplicates from Sorted Array II/80. Remove Duplicates from Sorted Array II.java
class Solution {

  public int removeDuplicates(int[] nums) {
    int n = nums.length;

    if (n < 3) {
      return n;
    }

    int left = 2;
    for (int right = 2; right < n; right++) {
      if (nums[right] != nums[left - 2]) {
        nums[left] = nums[right];
        left++;
      }
    }

    return left;
  }
}
